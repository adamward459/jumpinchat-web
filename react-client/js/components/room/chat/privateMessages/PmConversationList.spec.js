/* global jest, expect, describe, it, beforeEach */

import React from 'react';
import { shallow } from 'enzyme';
import PmConversationList from './PmConversationList.react';

describe('<PmConversationList />', () => {
  let props;
  beforeEach(() => {
    props = {
      selectedConversation: 'convo',
      privateMessages: [],
      selectConversation: jest.fn(),
      openMenu: jest.fn(),
      closeConversation: jest.fn(),
      handleClickOutside: jest.fn(),
    };
  });

  it('should render list of conversations', () => {
    props.privateMessages = [
      {
        user: { userListId: 'foo' },
        unreadMessages: 0,
      },
      {
        user: { userListId: 'bar' },
        unreadMessages: 0,
      },
    ];

    const wrapper = shallow(<PmConversationList {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('should highlight current conversation', () => {
    props.selectedConversation = 'foo';
    props.privateMessages = [
      {
        user: { userListId: 'foo' },
        unreadMessages: 0,
      },
      {
        user: { userListId: 'bar' },
        unreadMessages: 0,
      },
    ];

    const wrapper = shallow(<PmConversationList {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
