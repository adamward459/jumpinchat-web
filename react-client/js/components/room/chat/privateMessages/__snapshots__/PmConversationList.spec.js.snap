// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PmConversationList /> should highlight current conversation 1`] = `
<ForwardRef
  className="privateMessages__ListWrapper chat__UserList"
  horizontal={false}
>
  <PmListItem
    closeConversation={[MockFunction]}
    disabled={false}
    handleClickOutside={[MockFunction]}
    menuOpen={null}
    openMenu={[MockFunction]}
    selectConversation={[MockFunction]}
    selectedConversation="foo"
    unreadMessages={0}
    user={
      Object {
        "userListId": "foo",
      }
    }
  />
  <PmListItem
    closeConversation={[MockFunction]}
    disabled={false}
    handleClickOutside={[MockFunction]}
    menuOpen={null}
    openMenu={[MockFunction]}
    selectConversation={[MockFunction]}
    selectedConversation="foo"
    unreadMessages={0}
    user={
      Object {
        "userListId": "bar",
      }
    }
  />
</ForwardRef>
`;

exports[`<PmConversationList /> should render list of conversations 1`] = `
<ForwardRef
  className="privateMessages__ListWrapper chat__UserList"
  horizontal={false}
>
  <PmListItem
    closeConversation={[MockFunction]}
    disabled={false}
    handleClickOutside={[MockFunction]}
    menuOpen={null}
    openMenu={[MockFunction]}
    selectConversation={[MockFunction]}
    selectedConversation="convo"
    unreadMessages={0}
    user={
      Object {
        "userListId": "foo",
      }
    }
  />
  <PmListItem
    closeConversation={[MockFunction]}
    disabled={false}
    handleClickOutside={[MockFunction]}
    menuOpen={null}
    openMenu={[MockFunction]}
    selectConversation={[MockFunction]}
    selectedConversation="convo"
    unreadMessages={0}
    user={
      Object {
        "userListId": "bar",
      }
    }
  />
</ForwardRef>
`;
